expression/builtin_other.go

```diff
@@ -103,14 +135,53 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 	return sig, nil
 }
 
+type baseInSig struct {
+	baseBuiltinFunc
+	nonConstArgs []Expression
+	hasNull      bool
+}
+
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
-	baseBuiltinFunc
+	baseInSig
+	hashSet map[int64]bool
```

Comment:
It's better to add a comment that the `bool` value in the map is used to identify whether the constant stored in `key` is signed or unsigned.